movie_search/media.py

```diff
@@ -1,113 +1,265 @@
+        pass
+
+    @abstractmethod
+    def store_data(self, data):
+        pass
+
+    @abstractmethod
+    def get_discover_data(self, **kwargs):
+        pass
+
+
+# Context class to utilize the strategies
+class MediaContext:
+    def __init__(self, service: MediaService):
+        self._service = service
     
-    return movie 
+    def set_service(self, service: MediaService):
+        self._service = service
+
+    def process_media(self, tmdb_id):
+        data = self._service.fetch_from_api(tmdb_id)
+        return self._service.store_data(data)
 
 
-def store_movie_genres(genres):
-    movie_genres = []
+class MovieStrategy(MediaService):
+    def fetch_from_api(self, tmdb_id):
+        movie_data = tmdb_api_obj.get_data_from_endpoint(f"/movie/{tmdb_id}")
+        video_data = tmdb_api_obj.get_data_from_endpoint(f"/movie/{tmdb_id}/videos")
+        return movie_data, video_data
 
-    for row in genres:
-        genre, created = Genre.objects.get_or_create(
-            name=row.get("name", ""),
-            genre_id=row.get("id", 0),
+    def store_data(self, data):
+        movie_data, video_data = data
+        # Store movie data
+        genres = movie_data.get("genres")
+        movie, created = Movie.objects.get_or_create(
+            tmdb_id=movie_data.get("id", 0),
+            title=movie_data.get("title", ""),
+            backdrop_path=movie_data.get("backdrop_path", ""),
+            tagline=movie_data.get("tagline", ""),
+            vote_count=movie_data.get("vote_count", 0),
+            vote_average=movie_data.get("vote_average", 0),
+            popularity=movie_data.get("popularity", 0),
+            release_date=movie_data.get("release_date", ""),
+            runtime=movie_data.get("runtime", 0),
+            production_company=movie_data.get("production_company", ""),
+            overview=movie_data.get("overview", ""),
+            budget=movie_data.get("budget", 0),
+            revenue=movie_data.get("revenue", 0),
+            homepage=movie_data.get("homepage", ""),
         )
-        movie_genres.append(genre)
+        if genres is not None:
+            movie_genres = self.store_genres(genres)
+
+        movie.genres.add(*movie_genres)
+
+        movie_recommendations = self.store_recommendations(movie.tmdb_id)
+        movie.recommendation.add(*movie_recommendations)
 
-    return movie_genres
+        videos = self.store_videos(movie, video_data)
 
+        return movie, videos
 
-def store_movie_videos(movie_obj):
-    video_data = tmdb_api.get_data_from_endpoint(f"/movie/{movie_obj.movie_id}/videos")
-    for video_data in video_data.get("results", []):
-        Video.objects.get_or_create(
-            movie=movie_obj,
-            name=video_data.get("name", ""),
-            key=video_data.get("key", ""),
+    def store_genres(self, genres):
+        movie_genres = []
+        for row in genres:
+            genre, created = MovieGenre.objects.get_or_create(
```

Comment:
consider a bulk create, but I think we have discussed this before and the fact that some genres might be created already made this complicated right?

Reply:
Yeah it was discussed before